ape_hardhat/provider.py

```diff
@@ -417,9 +476,16 @@ def _start(self):
                             f"Unable to find an available port. Ports tried: {ports_str}"
                         )
 
-                self.port = port
+                self.attempted_ports.append(port)
+                self._host = f"http://127.0.0.1:{port}"
+
+        elif ":" in self._host and self._port is not None:
```

Comment:
so if i set the `host` to `127.0.01:8889`, it does not start the process on port 8889, it still uses the default port. I think because this line right here is wrong. It is saying both ":" is in the host AND the legacy port was set, which can never both be true.

So this needs to be fixed somehow.. the port should be used from host, may need a `def port()` property that parses and returns it when `_port` is not set.

In the `ape-foundry` PR, it seems a little different, may want to copy that.

But to parse the host is like:

```python
def _port(self) -> int:
  return URL(self._host).port
```

Reply:
The problem turned out to be that the port wasn't being parsed correctly from the `self.config.host` if the host was missing the `http://` in the `ape-config.yaml`.